packages/app/src/elements/import-dialog.ts

```diff
@@ -149,12 +150,22 @@ export class ImportDialog extends Dialog<File, void> {
                                     ?hidden=${itemColumn.type === "name" || itemColumn.type === "tags"}
                                 >
                                     <div class="small margined spacing horizontal layout">
-                                        <div class="stretch">${itemColumn.name}</div>
-                                        <div class="subtle">${$l("Column {0}", itemColumnIndex.toString())}</div>
+                                        <div class="stretch">
+                                            ${csvHasColumnsOnFirstRow
+                                                ? itemColumn.name
+                                                : $l("Column {0}", itemColumnIndex.toString())}
```

Comment:
If the first row is not column names, we can't expect meaningful/unique names here so better to just show the column number.